neurom/apps/morph_stats.py

```diff
@@ -164,17 +166,17 @@ def extract_stats(neurons, config):
     {config_path}
     """
 
-    def _fill_stats_dict(data, stat_name, stat):
+    def _fill_stats_dict(data, stat_name, stat, feature_name):
         """Insert the stat data in the dict.
 
         And if the stats is a 3D array, splits it into XYZ components.
         """
-        if stat is None or not isinstance(stat, np.ndarray) or stat.shape not in ((3, ), ):
-            data[stat_name] = stat
-        else:
+        if _find_feature_func(feature_name).shape[-1] == 3:
             for i, suffix in enumerate('XYZ'):
                 compound_stat_name = stat_name + '_' + suffix
-                data[compound_stat_name] = stat[i]
+                data[compound_stat_name] = stat[i] if stat is not None else None
+        else:
+            data[stat_name] = stat
```

Comment:
Just to be sure, do we need to do anything special also when shape is `(..., 2)` in the feature `partition_pairs`?

Reply:
Yes. Good point.